tests/languages/jsx!+graphql+js-templates/graphql_inclusion.test

```diff
@@ -0,0 +1,223 @@
+
+// GraphQL query to retrieve products by IDs.
+// The price field belongs to the variants object because
+// variations of a product can have different prices.
+const GET_PRODUCTS_BY_ID = gql`
+  query getProducts($ids: [ID!]!) {
+    nodes(ids: $ids) {
+      ... on Product {
+        title
+        handle
+        descriptionHtml
+        id
+        images(first: 1) {
+          edges {
+            node {
+              originalSrc
+              altText
+            }
+          }
+        }
+        variants(first: 1) {
+          edges {
+            node {
+              price
+              id
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+----------------------------------------------------
+
+[
+	["keyword", "import"],
+	" React ",
+	["keyword", "from"],
+	["string", "'react'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	" gql ",
+	["keyword", "from"],
+	["string", "'graphql-tag'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	["punctuation", "{"],
+	" Query ",
+	["punctuation", "}"],
+	["keyword", "from"],
+	["string", "'react-apollo'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	["punctuation", "{"],
+
+	"\r\n  Card",
+	["punctuation", ","],
+
+	"\r\n  ResourceList",
+	["punctuation", ","],
+
+	"\r\n  Stack",
+	["punctuation", ","],
+
+	"\r\n  TextStyle",
+	["punctuation", ","],
+
+	"\r\n  Thumbnail",
+	["punctuation", ","],
+
+	["punctuation", "}"],
+	["keyword", "from"],
+	["string", "'@shopify/polaris'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	" store ",
+	["keyword", "from"],
+	["string", "'store-js'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	["punctuation", "{"],
+	" Redirect ",
+	["punctuation", "}"],
+	["keyword", "from"],
+	["string", "'@shopify/app-bridge/actions'"],
+	["punctuation", ";"],
+
+	["keyword", "import"],
+	["punctuation", "{"],
+	" Context ",
+	["punctuation", "}"],
+	["keyword", "from"],
+	["string", "'@shopify/app-bridge-react'"],
+	["punctuation", ";"],
+
+	["comment", "// GraphQL query to retrieve products by IDs."],
+
+	["comment", "// The price field belongs to the variants object because"],
+
+	["comment", "// variations of a product can have different prices."],
+
+	["keyword", "const"],
+	["constant", "GET_PRODUCTS_BY_ID"],
+	["operator", "="],
+	" gql",
+	["template-string", [
+		["template-punctuation", "`"],
+		["graphql", [
+			["keyword", "query"],
+			["definition-query", "getProducts"],
+			["punctuation", "("],
+			["variable", "$ids"],
+			["punctuation", ":"],
+			["punctuation", "["],
+			["scalar", "ID"],
+			["operator", "!"],
+			["punctuation", "]"],
+			["operator", "!"],
+			["punctuation", ")"],
+			["punctuation", "{"],
+
+			["property-query", "nodes"],
+			["punctuation", "("],
+			["attr-name", "ids"],
+			["punctuation", ":"],
+			["variable", "$ids"],
+			["punctuation", ")"],
+			["punctuation", "{"],
+
+			["operator", "..."],
+			["keyword", "on"],
```

Comment:
@RunDevelopment I noticed this would be tokenized to `definition-mutation` while I was working on the template string issue (#2957)